### PR TITLE
basics of js 

### PR DESCRIPTION
**Pull request by Sakshi-Prajapat**

### CODE REVIEW

File: Arr.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+///////////// find = returns first element which passes the condition 
+// let found = arr.find(function (element) {
+//     return element > 10;
+// });
+// let indexOf = arr.indexOf(found);  /// returns the indexof element
+// console.log(indexOf)
+
+/// Filter = it return the array of all element that passes the condition
+
```
Comment (Ritikthoughtwin) [inline Arr.js:19]: test-comment

File: Arr.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+//     console.log("geeks",x);
+// }
+
+///////////// find = returns first element which passes the condition 
+// let found = arr.find(function (element) {
+//     return element > 10;
+// });
+// let indexOf = arr.indexOf(found);  /// returns the indexof element
+// console.log(indexOf)
+
```
Comment (Ritikthoughtwin) [inline Arr.js]: Why all this code is commented?

File: Arr.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+ let modifyArr = arr.splice(2,1, 1000,500)
+ console.log("spice",modifyArr)
+    //  let removed = languages.splice(2, 1, 'Julia', 'Php');
+  webDvlop.splice(2, 1, 'PHP', 'React_Native')
+     console.log(webDvlop)
+
+let arr1 = ['apple', 'banana', 'cherry', 'dates', 'elderberry'];
+
+//  arr1.splice(2, 1, 'fig', 'grapes');
+ let newarr =  arr1.splice(2, 1, 'fig', 'grapes');
```
Comment (Ritikthoughtwin) [inline Arr.js]: All the naming conventions should be proper and informative

File: hoisting.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+const myObject = {
+    regularExample: function() {
+      console.log("REGULAR: ", this)
+    },
+  
+    arrowExample: () => {
+      console.log("ARROW: ", this)
+    }
+  }
+
```
Comment (Ritikthoughtwin) [inline hoisting.js]: Maintain proper naming conventions

File: promise.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    if (x === y) {
+        resolve();
+    } else {
+        reject();
+    }
+});
+
+promise.
+    then(function () {
+        console.log('Success, You are a GEEK');
```
Comment (Ritikthoughtwin) [inline promise.js]: Add an example of arrow function too

File: string.js [hunk 6]
```diff
@@ -0,0 +1,6 @@
+let str = "helloworld"
+// console.log(str.split(""))// 'h', 'e', 'l', 'l','o', 'w', 'o', 'r','l', 'd'
+// console.log(str.split(','))
+
+sl=str.slice(0,5)
+console.log(sl)
```
Comment (Ritikthoughtwin) [inline string.js]: Fix the naming conventions.

File: Arr.js [hunk 7]
```diff
@@ -0,0 +1,57 @@
+const arr = [33,56,78,66,22,88,66]
+let languages = ['C++', 'Java', 'Html', 'Python', 'C'];
+let webDvlop = ["HTML", "CSS", "JS", "Bootstrap"];
+
+//////////// entries = return key value pair of every index
+// let v = languages.entries();
+// for(x of v ){
```
Comment (Ritikthoughtwin) [inline Arr.js]: Variable and Iterator names should be informative.

File: Arr.js [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+console.log(arr.reduce(sumArr));
+let new_arr = arr.slice(2, 6);
+console.log(new_arr);
+ let modifyArr = arr.splice(2,1, 1000,500)
+ console.log("spice",modifyArr)
+webDvlop.splice(2, 1, 'PHP', 'React_Native')
+console.log(webDvlop)
+
+let fruits = ['apple', 'banana', 'cherry', 'dates', 'elderberry'];
+
```
Comment (Ritikthoughtwin) [inline Arr.js]: Use only one declaration for multiple uses.

File: asyncAwait.js [hunk 9]
```diff
@@ -0,0 +1,14 @@
+const test = () => {
+    return "test"
+}
+
+async function  hello(){
+    //  console.log("hello")
+    const res =  test();
+    return res
+}
+
```
Comment (Ritikthoughtwin) [inline asyncAwait.js:9]: Add arrow function example as well.

File: callback.js [hunk 10]
```diff
@@ -0,0 +1,13 @@
+function first(){
+    console.log("hello1")
+}
+
+function second(callback){
+    setTimeout(()=>{
+        callback()
+    },2000)
+    console.log("world")
```
Comment (Ritikthoughtwin) [inline callback.js:10]: code should be well indented and decalrations should be proper

File: Arr.js [hunk 11]
```diff
@@ -0,0 +1,63 @@
+const arr = [33,56,78,66,22,88,66]
+let languages = ['C++', 'Java', 'Html', 'Python', 'C'];
+let webDvlop = ["HTML", "CSS", "JS", "Bootstrap"];
+
+//////////// entries = return key value pair of every index
+let v = languages.entries();
```
Comment (tw-vineet) [inline Arr.js]: entries is not a array method please check and explain ?

**Comment by tw-vineet** [comment]
file `Arr `name rename to `arrayMethod.`

File: Arr.js [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+for(x of v ){
+    console.log("entries",x);
```
Comment (Ritikthoughtwin) [inline Arr.js]: Declaration and iterator naming conventions should be proper and informative.

File: Arr.js [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+function sumArr(sum , num){
+    return sum+num;
+}
+console.log(arr.reduce(sumArr));
+let new_arr = arr.slice(2, 6);
+console.log(new_arr);
+ let modifyArr = arr.splice(2,1, 1000,500)
+ console.log("spice",modifyArr)
+webDvlop.splice(2, 1, 'PHP', 'React_Native')
+console.log(webDvlop)
```
Comment (tw-vineet) [inline Arr.js]: Maintain code in formatted.

File: promise.js [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+
+promise.
+    then(function () {
+        console.log('Success with regular function');
+    }).
+    catch(function () {
+        console.log('Some error has occurred');
+    }); 
+
+
```
Comment (Ritikthoughtwin) [inline promise.js:58]: Seems you are copy-pasting the code from outsourced websites.

**Comment by tw-vineet** [comment]
Add definitions based on all topics

File: asyncAwait.js (hunk 9 above)
Comment (Sakshi-Prajapat) [reply asyncAwait.js:9]: add an example of arrow function

File: arrayMethod.js [hunk 15]
```diff
@@ -0,0 +1,72 @@
...
+
+/// Filter = it return the array of all element that passes the condition
+
+let filtered = arr.filter(isEven);
+console.log("filter",filtered)
+
+
+
+///////////////////////////some method :- pass a test and return a boolean
+function isEven(value) {
```
Comment (Ritikthoughtwin) [inline arrayMethod.js:27]: No need to add multiple forward slash "/" only two is sufficient to comment any line.

File: arrayMethod.js [hunk 16]
```diff
@@ -0,0 +1,72 @@
...
+    }
+    
+})
+console.log(newMapArray)
+let forEachArr = languages.forEach((item)=>{
+     console.log("froEach",item)
+   
+})
+
+let filterPro = arr.filter((item)=>{
```
Comment (Ritikthoughtwin) [inline arrayMethod.js]: Fix naming conventions.

File: arrayMethod.js [hunk 17]
```diff
@@ -0,0 +1,72 @@
...
+
+function sumArr(sum , num){
+    return sum+num;
+}
+console.log(arr.reduce(sumArr));
+let new_arr = arr.slice(2, 6);
+console.log(new_arr);
+ let modifyArr = arr.splice(2,1, 1000,500)
+ console.log("spice",modifyArr)
+webDvlop.splice(2, 1, 'PHP', 'React_Native')
```
Comment (Ritikthoughtwin) [inline arrayMethod.js]: Write correct spelling.

File: promise.js [hunk 18]
```diff
@@ -0,0 +1,63 @@
...
+    const y = 1;
+    if (x === y) {
+        resolve(4);
+    } else {
+        reject();
+    }
+});
+
+let promise1 =  Promise.resolve(6)
+let promise2 = new Promise(function(resolve,){
```
Comment (Ritikthoughtwin) [inline promise.js]: variable names should not contain numerics.

Comment (Sakshi-Prajapat) [reply promise.js]: change the variable name

File: arrayMethod.js (hunk 16 above)
Comment (Sakshi-Prajapat) [reply arrayMethod.js]: fixed the naming conventions

File: arrayMethod.js [hunk 19]
```diff
@@ -0,0 +1,72 @@
...
+        return item;
+    }
+    
+})
+console.log(newMapArray)
+let forEachArr = languages.forEach((item)=>{
+     console.log("froEach",item)
+   
+})
+
```
Comment (Ritikthoughtwin) [inline arrayMethod.js:68]: ForEachArr fix naming conventions.
Can we assign a loop to any variable?
can't see the output for this loop. what will be the output?

Comment (Ritikthoughtwin) [inline arrayMethod.js:68]: write an example using all loops.

File: asyncAwait.js [hunk 20]
```diff
@@ -0,0 +1,21 @@
+const test = () => {
+    return "test"
+}
+
+async function  hello(){
```
Comment (Ritikthoughtwin) [inline asyncAwait.js]: Fix naming conventions

File: hoisting.js [hunk 21]
```diff
@@ -0,0 +1,12 @@
+const myObject = {
+    regularFunction: function() {
+      console.log("REGULAR FUNCTION: ", this)
+    },
+
```
Comment (Ritikthoughtwin) [inline hoisting.js]: Add an example of using this keyword here. add references for it.

File: promise.js [hunk 22]
```diff
@@ -0,0 +1,63 @@
...
+    const x = 1;
+    const y = 1;
+    if (x === y) {
+        resolve(4);
+    } else {
+        reject();
+    }
+});
+
+let promiseTwo =  Promise.resolve(6)
```
Comment (Ritikthoughtwin) [inline promise.js]: Use const if you are not updating the value of variables.

**Comment by Ritikthoughtwin** [comment]
Add examples for closures as well.

File: string.js [hunk 23]
```diff
@@ -0,0 +1,14 @@
...
+sliceString=string.slice(0,5)
+console.log(sliceString)
+
+console.log("Upper Case",string.toUpperCase())
+
+console.log("Char at index",string.charAt(7))
+
+console.log("unicode at index",string.charCodeAt(8))
+
+console.log("unicode at index",string.codePointAt(8))
```
Comment (Ritikthoughtwin) [inline string.js:14]: Explore more string methods and add examples for them.

File: callback.js [hunk 24]
```diff
@@ -0,0 +1,24 @@
+let num = [1,2,3,4,5]
```
Comment (Ritikthoughtwin) [inline callback.js]: what is num? here.

File: Arr.js (hunk 11 above)
Comment (Sakshi-Prajapat) [reply Arr.js]: I checked , entries is an array method here is the reference :- https://www.geeksforgeeks.org/javascript-array-entries-method/

File: arrayMethod.js (hunk 19 above)
Comment (Sakshi-Prajapat) [reply arrayMethod.js:68]: fixed the name of variable and if we assign a loop to any variable it will give undefine

**Comment by Ritikthoughtwin** [comment]
LGTM for now.

**Review by Ritikthoughtwin — APPROVED** [review]

**Pull request merged**